src/software/gui/full_system/widgets/ai_control.cpp

```diff
@@ -11,63 +11,116 @@
-        else if (text == "Use Refbox")
-        {
-            sensor_fusion_config->mutableOverrideRefboxFriendlyTeamColor()->setValue(
-                false);
-        }
-        else
-        {
-            LOG(FATAL) << "Tried to set the team colour with an invalid value: '"
-                       << text.toStdString() << "'" << std::endl;
-        }
-    };
+    auto on_team_colour_changed =
+        [override_friendly_colour_parameter,
+         friendly_colour_yellow_parameter](const QString &text) {
+            if (text == "Yellow")
+            {
+                override_friendly_colour_parameter->setValue(true);
+                friendly_colour_yellow_parameter->setValue(true);
+            }
+            else if (text == "Blue")
+            {
+                override_friendly_colour_parameter->setValue(true);
+                friendly_colour_yellow_parameter->setValue(false);
+            }
+            else if (text == "Use Refbox")
+            {
+                override_friendly_colour_parameter->setValue(false);
+            }
+            else
+            {
+                LOG(FATAL) << "Tried to set the team colour with an invalid value: '"
+                           << text.toStdString() << "'" << std::endl;
+            }
+        };
     QWidget::connect(widget->team_colour_combo_box, &QComboBox::currentTextChanged,
                      on_team_colour_changed);
 }
 
-void setupGameStateOverrideComboBox(Ui::AutogeneratedFullSystemMainWidget *widget,
-                                    std::shared_ptr<ThunderbotsConfig> config)
+void setupDefendingSideComboBox(
+    Ui::AutogeneratedFullSystemMainWidget *widget,
+    std::shared_ptr<Parameter<bool>> defending_side_override_parameter,
+    std::shared_ptr<Parameter<bool>> defending_positive_side_parameter)
+{
+    // See issue #811 for getting these value from an enum / factory
+    widget->defending_side_combo_box->insertItem(0, "Use Refbox");
+    widget->defending_side_combo_box->insertItem(1, "East");
+    widget->defending_side_combo_box->insertItem(2, "West");
+
+    auto on_defending_side_changed =
+        [defending_side_override_parameter,
+         defending_positive_side_parameter](const QString &text) {
+            if (text == "Use Refbox")
+            {
+                defending_side_override_parameter->setValue(false);
+            }
+            else if (text == "Positive")
+            {
+                defending_side_override_parameter->setValue(true);
+                defending_positive_side_parameter->setValue(true);
+            }
+            else if (text == "Negative")
+            {
+                defending_side_override_parameter->setValue(true);
+                defending_positive_side_parameter->setValue(false);
+            }
+            else
+            {
+                LOG(FATAL) << "Tried to set the defending side with an invalid value: '"
+                           << text.toStdString() << "'" << std::endl;
+            }
```

Comment:
Ran locally

~~~
2020/07/06 21:06:48 215780      FATAL [ai_control.cpp->operator():111]                                                                                                                                                                     
        *******  EXIT trigger caused by LOG(FATAL) entry:                                                                                                                                                                                  
        "Tried to set the defending side with an invalid value: 'East'   
~~~

This currently triggers because the options currently are `East` and `West` and not `Positive` and `Negative`

Reply:
Heh whoops :sweat_smile:

Reply:
fixed